test/asg.tf

```diff
@@ -0,0 +1,29 @@
+module "asg-test" {
+  source                  = "../aws/asg"
+  app_name                = "test_name1"
+  app_env                 = "test_env1"
+  ami_id                  = "ami-1234567890"
+  aws_instance            = { instance_type = "t3.nano", volume_size = "8", instance_count = "1" }
+  private_subnet_ids      = ["1234", "5678"]
+  default_sg_id           = "sg-asdf"
+  ecs_instance_profile_id = "asdf"
+  ecs_cluster_name        = "mycluster1"
```

Comment:
Do you want to include `cpu_credits` in one of these?

Reply:
I assumed we were only including the *required* inputs.  Is that a bad assumption?  Should we include the other optional inputs as well?  If some are mutually exclusive, do we create separate tests for the valid combinations?

Reply:
@briskt Do you have an opinion on this?